### PR TITLE
Core/BG: implement the rated bg ui opcode

### LINKED ISSUE #68: Cannot join the RatedBG queue
**Description:**

There's no way to queue Rated Battlegrounds.

**Current behaviour:**

Currently in the PVP UI, in conquest section, there's no way to queue Rated Battlegrounds, even in a 10 man raid party (as required) the button remains disabled and the UI shows no change (the group is still mark as invalid).

**Expected behaviour:**

On a 10 man raid, the join button should be available.

**Steps to reproduce the problem:**

1. Run 10 clients.
2. Join as a raid group.
3. The raid leader opens the UI, the group should be marked as invalid and the button will remain disabled.

**TC rev. hash/commit:** 

TrinityCore rev. d49579395ef3 2020-04-22 18:17:17 +0200 (master branch) (Unix, RelWithDebInfo, Static) (worldserver-daemon)

**TDB version:** TDB_full_world_434.20021_2020_02_21.sql

**Operating system:** Linux Mint 19.04

### PR DESCRIPTION
**Pull request by kenpb**
This will allow to toggle the rated battleground UI for the player.

**Changes proposed**:

- Send a worldstate 5508 after sending the SMSG_BATTLEFIELD_RATED_INFO (sniffs shown it as the next package received after)
- Add configuration for the rated bg reward amount and a enable/disable flag

**Issues addressed**: Fixes #68 

**Tests performed**: Tested in-game

**Known issues and TODO list**:

-  The RBG queue/system is yet to be implemented.

### CODE REVIEW

File: src/server/game/Battlegrounds/Battleground.h [hunk 1]
```diff
@@ -234,6 +234,7 @@ enum BGHonorMode
 
 #define BG_AWARD_ARENA_POINTS_MIN_LEVEL 71
 #define ARENA_TIMELIMIT_POINTS_LOSS    -16
+#define RATED_BATTLEGROUND_STATE_ACTIVE 5508
```
Comment (Ovahlord) [inline src/server/game/Battlegrounds/Battleground.h:237]: Put WS_ behind that enum to clarify that it's a world state to follow the remaining world state enum code style

File: src/server/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -2662,6 +2663,26 @@ Battleground.ReportAFK = 3
...
+#                     1 - (Enabled, Experimental as in not implemented)
+
+RatedBattleground.Enable = 0
+
+#
+#    RatedBattleground.Reward
+#        Description: The conquest points reward after winning a rated battleground.
+#        Default:     400
+
+RatedBattleground.Reward = 400
```
Comment (Ovahlord) [inline src/server/worldserver/worldserver.conf.dist:2681]: conquest points use currency precision so I personally would prefer using 40000 instead to reflect that so we have perfect control over the value.

**Comment by kenpb** [comment]
Hello @Ovahlord,

Thanks for taking the time to review the PR, I've corrected the issues you mentioned, let me know if I'm missing something.

Cheers.

**Comment by Ovahlord** [comment]
You haven't done my previous request yet. Do that and it's good to go.

**Comment by kenpb** [comment]
Sorry, can you elaborate? I thought I did

**Comment by Ovahlord** [comment]
See what I commented at the worldserver config. The enable / disable option is not really a boolean value. It's a number for the world state (0 = disabled, 1 = 10v10, 2 = 15v15, 3 = 25v25)

**Comment by kenpb** [comment]
Hello @Ovahlord, sorry for the (extremely) late response, I see, from the single sniff I had I couldn't pick this up, thanks for the heads up!

I will continue to look into the implementation of rated bgs, do you, by any chance, know where can I get more sniffs?, this is pretty much what I have and by looking at the wpp I know there's a different structure for a rated scoreboard for example.

**Comment by Ovahlord** [comment]
Aight. Looks good from my side. anything left to add?

**Comment by kenpb** [comment]
Cool, and not at the moment, I'll start another PR when I continue

**Pull request merged**